lib/Doctrine/Common/DataFixtures/Purger/ORMPurger.php

```diff
@@ -129,40 +145,43 @@ public function purge()
             $orderedTables[] = $this->getTableName($class, $platform);
         }
 
-        $connection            = $this->em->getConnection();
-        $filterExpr            = method_exists(
-            $connection->getConfiguration(),
+        $connectionConfiguration = $connection->getConfiguration();
+
+        $filterExpr = method_exists(
+            $connectionConfiguration,
             'getFilterSchemaAssetsExpression'
-        ) ? $connection->getConfiguration()->getFilterSchemaAssetsExpression() : null;
-        $emptyFilterExpression = empty($filterExpr);
+        ) ? $connectionConfiguration->getFilterSchemaAssetsExpression() : null;
 
         $schemaAssetsFilter = method_exists(
-            $connection->getConfiguration(),
+            $connectionConfiguration,
             'getSchemaAssetsFilter'
-        ) ? $connection->getConfiguration()->getSchemaAssetsFilter() : null;
+        ) ? $connectionConfiguration->getSchemaAssetsFilter() : null;
 
+        $this->cachedSqlStatements = [];
         foreach ($orderedTables as $tbl) {
             // If we have a filter expression, check it and skip if necessary
-            if (! $emptyFilterExpression && ! preg_match($filterExpr, $tbl)) {
+            if (! empty($filterExpr) && ! preg_match($filterExpr, $tbl)) {
```

Comment:
Why is inlining `$emptyFilterExpression` into the loop a performance improvement? `empty($filterExpr)` should be an invariant expression, why evaluate it on each iteration?

Reply:
This change is not about performance, but to fix IDE error. I don't think it can impact much
![image](https://user-images.githubusercontent.com/661654/197268642-69076566-66f8-47e5-bcff-b03574ecb4a7.png)

Reply:
In that case, I suggest to revert it.

Reply:
Done, but I reverted a condition from `!$emptyFilterExpression` to `$hasFilterExpression` - easy to read